cpp/src/lists/lists_column_factories.cu

```diff
@@ -16,10 +16,62 @@
 
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_factories.hpp>
+#include <cudf/detail/gather.cuh>
 
 #include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/sequence.h>
 
 namespace cudf {
+namespace lists {
+namespace detail {
+
+std::unique_ptr<cudf::column> make_lists_column_from_scalar(list_scalar const& value,
+                                                            size_type size,
+                                                            rmm::cuda_stream_view stream,
+                                                            rmm::mr::device_memory_resource* mr)
+{
+  if (size == 0) {
+    return make_lists_column(0,
+                             make_empty_column(data_type{type_id::INT32}),
+                             empty_like(value.view()),
+                             0,
+                             cudf::create_null_mask(0, mask_state::UNALLOCATED));
+  }
+  // Handcraft a 1-row column
+  auto offsets   = make_numeric_column(data_type(type_id::INT32), 2, mask_state::UNALLOCATED);
```

Comment:
Similar to what @harrism said about mask, the offsets should also be allocated using the appropriate `mr` based on whether it will be included in the returned 1 row column or not.